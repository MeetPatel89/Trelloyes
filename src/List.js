import React from 'react';
import Card from './Card';
import Store from './Store';


 function List (props) {
    
    
const cardLists = props.cardIds.map(function(cardId) {
    let card = Store.allCards[cardId]
    return <Card key={cardId} title={card.title} content={card.content}
    />
})

   return (
       
    <section className="List">
        <header className="List-header">
        <h2>{props.header}</h2>
        </header>
        <div className="List-cards">
        </div>
        {cardLists}
    </section>

   )
}


export default List;

