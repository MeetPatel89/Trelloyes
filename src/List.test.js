import React from 'react';
import ReactDOM from 'react-dom';

import List from './List';
import Store from './Store';
import renderer from 'react-test-renderer';


describe('<List />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        // Pass required props in the usual way
        ReactDOM.render(
            <List
            key={'3'} header={'Third list'} cardIds={[ 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm' ]}
                
            
            />,
            div
        );

        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<List header="Second list" cardIds={["b", "c", "d", "f", "h", "i", "k"]}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});;