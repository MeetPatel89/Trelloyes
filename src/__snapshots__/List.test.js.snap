// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Second list
    </h2>
  </header>
  <div
    className="List-cards"
  />
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Second card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Third card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Fourth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Sixth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Eighth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Ninth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Eleventh card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>
</section>
`;
